azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -185,12 +185,13 @@ class PipelineRootStage(PipelineStage):
     :type on_disconnected_handler: Function
     """
 
-    def __init__(self):
+    def __init__(self, pipeline_configurations=None):
         super(PipelineRootStage, self).__init__()
         self.on_pipeline_event_handler = None
         self.on_connected_handler = None
         self.on_disconnected_handler = None
         self.connected = False
+        self.pipeline_configurations = pipeline_configurations
```

Comment:
>pipeline_configurations [](start = 39, length = 23)

nit: "configurations" with an s implies that this is possibly more than one configuration object.  My first thought is that this holds some set of possible configurations and there must be some way to choose between them.

Reply:
done